src/core/model/Document.cpp

```diff
@@ -32,8 +33,7 @@ Document::~Document() {
 
 void Document::freeTreeContentModel() {
     if (this->contentsModel) {
-        gtk_tree_model_foreach(this->contentsModel.get(),
-                               reinterpret_cast<GtkTreeModelForeachFunc>(freeTreeContentEntry), this);
+        gtk_tree_model_foreach(this->contentsModel.get(), xoj::util::wrap_for_once<freeTreeContentEntry>(), this);
```

Comment:
This is weird. The fact that G_SOURCE_REMOVE coincides with "do not break the foreach" is a coincidence.
```suggestion
        gtk_tree_model_foreach(this->contentsModel.get(), xoj::util::wrap<freeTreeContentEntry>(), this);
```

Reply:
good catch!